### PR TITLE
edit.php SQL statement done

### PR DESCRIPTION
**Pull request by Osvy789**
Add.php done and edit.php almost done

### CODE REVIEW

**Comment by timothyoei** [comment]
Can you split this commit so the endpoints that don't work (i.e., edit and delete) aren't added yet? We don't want to merge unfinished code into main as it's our "working" copy

File: api/add.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+$dbConn = APIUtils::connectToDB();
+
+// Check if connection attempt failed
+if ($dbConn->connect_error)
+{
+  APIUtils::sendJsonResultInfo($rows, $endpoint, APIUtils::$failStatus, "Could Not Connect To Database");
+}
+else
+{
+    $addSQL = $dbConn->prepare("INSERT INTO Contacts (LastName, FirstName, Email, Phone, Address, Zip, City, State, ContactID, UserID) VALUES ('" . $LastName . "','" . $FirstName . "','" . $Email . "','" . $Phone . "','" . $Address . "','" . $Zip . "','" . $City . "','" . $State . "','" . $ContactID . "','" . $UserID . "')");
```
Comment (timothyoei) [inline api/add.php:33]: Appending a bunch of values like this into a one-liner isn't very readable. Can you refactor this?

File: api/add.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// Access util methods
+require_once("api_utils.php");
+
+// Set up return arguments
+$rows = [];
+$endpoint = "add";
+
+// Get input data
+$inData = APIUtils::getRequestInfo();
+$LastName = $inData['LastName'];
```
Comment (timothyoei) [inline api/add.php:12]: We only access these properties once in the sql statement so no need to create variables for each one

**Review by timothyoei — CHANGES_REQUESTED** [review]

File: api/delete.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// Try to connect to the database
+$dbConn = APIUtils::connectToDB();
+
+// Check if connection attempt failed
+if ($dbConn->connect_error)
+{
+  APIUtils::sendJsonResultInfo($rows, $endpoint, APIUtils::$failStatus, "Could Not Connect To Database");
+}
+else{
+    $deleteSQL = $dbConn->prepare("DELETE FROM Contacts WHERE contactID= '" . $ContactID . "', UserID='" . $UserID . "'");
```
Comment (timothyoei) [inline api/delete.php:24]: Does this work on the server? You might need to add an AND in that WHERE

File: api/delete.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}
+try {
+    if ($deleteSQL->execute()) {
+      APIUtils::sendJsonResultInfo($rows, $endpoint, APIUtils::$successStatus);
+    } else {
+      throw new Exception("Could Not Insert Record");
```
Comment (timothyoei) [inline api/delete.php:30]: Change this error message to be more descriptive of what failed?

File: api/edit.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+// Try to connect to the database
+$dbConn = APIUtils::connectToDB();
+
+// Check if connection attempt failed
+if ($dbConn->connect_error)
+{
+  APIUtils::sendJsonResultInfo($rows, $endpoint, APIUtils::$failStatus, "Could Not Connect To Database");
+}
+else{
+    $editSQL = $dbConn->prepare("UPDATE Contacts SET LastName='" . $LastName . "', FirstName= '" . $FirstName . "', Email= '" . $Email . "', Phone='" . $Phone . "', Address= '" . $Address . "', Zip= '" . $Zip . "', City= '" . $City . "', State= '" . $State . "' WHERE contactID= '" . $ContactID . "', UserID='" . $UserID ."'");
```
Comment (timothyoei) [inline api/edit.php:33]: Same as above comments

File: api/edit.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+}
+try {
+    if ($editSQL->execute()) {
+      APIUtils::sendJsonResultInfo($rows, $endpoint, APIUtils::$successStatus);
+    } else {
+      throw new Exception("Could Not Insert Record");
```
Comment (timothyoei) [inline api/edit.php:39]: Same as previous error comment

**Review by timothyoei — CHANGES_REQUESTED** [review]

File: api/search.php [hunk 7]
```diff
@@ -21,7 +21,7 @@
 else
 {
   // Prepare SQL statement for search attempt
-  $searchSQL = $dbConn->prepare("SELECT firstName, lastName FROM Users WHERE firstName LIKE ? OR lastName LIKE ?");
+  $searchSQL = $dbConn->prepare("SELECT firstName, lastName FROM Contacts WHERE firstName LIKE ? OR lastName LIKE ?");
```
Comment (timothyoei) [inline api/search.php:24]: Fixed in this [pr](https://github.com/timothyoei/cop4331_small_project/pull/14)

File: api/edit.php [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+// Check if connection attempt failed
+if ($dbConn->connect_error)
+{
+  APIUtils::sendJsonResultInfo($rows, $endpoint, APIUtils::$failStatus, "Could Not Connect To Database");
+}
+else{
+    $editSQL = $dbConn->prepare("UPDATE Contacts SET (LastName='" . $LastName . "', FirstName= '" . $FirstName . "', 
+                                Email= '" . $Email . "', Phone='" . $Phone . "',
+                                Address= '" . $Address . "', Zip= '" . $Zip . "', City= '" . $City . "', State= '" . $State . "') 
+                                AND WHERE contactID= '" . $ContactID . "', UserID='" . $UserID ."'");
```
Comment (timothyoei) [inline api/edit.php:36]: AND goes inside the WHERE (e.g., WHERE ContactID = $ContactID AND UserID = $UserID) <--- NOT the solution

**Review by timothyoei — CHANGES_REQUESTED** [review]

**Comment by timothyoei** [comment]
@Osvy789 Can you add a test plan/screenshots of this code working on the server with postman? Take a look at the previous api pr's for examples

**Comment by timothyoei** [comment]
Let's wait until after the demo to make changes to the api. You'll also need to fix the merge conflict as #14 went in

File: api/add.php (hunk 2 above)
Comment (Osvy789) [reply api/add.php]: I know I just did it because it helps me in the implementing the variables at least it makes it easier.

**Review by timothyoei — APPROVED** [review]

**Review by timothyoei — APPROVED** [review]

**Review by timothyoei — APPROVED** [review]

**Pull request merged**